testdata/gapi/onnx/download_onnx_models.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python
+
+import sys
+import os
+from shutil import rmtree as rmdir_with_data
+import hashlib
+
+CUR_DIR = os.getcwd()
+# This directory contains result of "git clone"
+CACHE_DIR = CUR_DIR + '/onnx_models_cache/'
+
+class Model:
+    MB = 1024*1024
+    BUFSIZE = 10*MB
+    # This directory contains large ONNX models that will be used
+    MODELS_DIR = CUR_DIR + '/onnx_models/'
+
+    def __init__(self, **kwargs):
+        self.name = kwargs.pop('name')
+        self.filepath = kwargs.pop('filepath')
+        self.sha = kwargs.pop('sha')
+        self.model_path = self.filepath + self.name + '.onnx'
+
+    def __str__(self):
+        return '[Model]: <{}>'.format(self.name + '\n          '+ self.filepath)
+
+    def sys_lfs_call(self):
+        return 'cd onnx_models_cache && git lfs pull --include=/' + str(self.model_path) + ' --exclude="" '
+
+    def verify(self, path):
+        if not os.path.exists(path):
+            return False
+        if not self.sha:
+            return False
+        print('[Info]: Verifying file:')
+        print('        Expected sha: {}'.format(self.sha))
+        sha = hashlib.sha1()
+        try:
+            with open(path, 'rb') as f:
+                while True:
+                    buf = f.read(self.BUFSIZE)
+                    if not buf:
+                        break
+                    sha.update(buf)
+            print('        Actual sha: {}'.format(sha.hexdigest()))
+            return self.sha == sha.hexdigest()
+        except Exception as e:
+            print('[Excn]: Catch {}'.format(e))
+
+    def create_dir(self):
+        model_dir_path = self.MODELS_DIR + self.filepath
+        if not os.path.exists(model_dir_path):
+            os.makedirs(model_dir_path)
+        else:
+            print('[Warn]: Directory already contains a folder for {} - skipping'.format(self.name))
+
+    def download(self):
+        print('______________________{}______________________'.format(self.name))
+        print('[Info]: Creating directory for {}'.format(self.name))
+        self.create_dir()
+        if self.verify(self.MODELS_DIR + self.model_path):
+            print('[Warn]: Hash match - skipping')
+            return True
+        print('[Info]: Downloading model')
+        # Pull large model file
+        os.system(self.sys_lfs_call())
+        if self.verify(CACHE_DIR + self.model_path):
+        # Move large model file to onnx_models
+            os.replace(CACHE_DIR + self.model_path, self.MODELS_DIR + self.model_path)
+            return True
+        return False
+
+def clear_pulled_files():
+    # "lfs" directory contains some data
+    # Remove this folder then "onnx_models_cache" will have standard size - 180 MB
+    # If we remove "onnx_models" folder then models will be downloaded again
+    # and "lfs" folder will exist during download
+    lfs_cache_path = CACHE_DIR + '.git/lfs'
+    if os.path.exists(lfs_cache_path):
+        rmdir_with_data(lfs_cache_path)
```

Comment:
Do we really need to clean '.git' internals? Does it break anything?

Reply:
No, but without cleaning models won't be downloaded again and will be moved from cache. And for models update you should remove cache directory.
It is main question about behavior. Do we need update models from GH when we remove models directory (NOT cache)?

Reply:
Don't clear the cache now.